pyglet/math.py

```diff
@@ -86,6 +86,31 @@ def __neg__(self):
     def __round__(self, ndigits=None):
         return Vec2(*(round(v, ndigits) for v in self))
 
+    def __radd__(self, other):
+        if other == 0:
+            return self
+        else:
+            return self.__add__(other)
+
+    def heading(self):
+        return _math.atan2(self[1], self[0])
+
+    def set_heading(self, angle):
+        mag = self.__abs__()
+        return Vec2(mag * _math.cos(angle), mag * _math.sin(angle))
```

Comment:
The name here is a bit confusing. `set_*` often implies we are setting some state on the instance. Also a docstring and an indication what this method returns.

Reply:
Currently Vectors are immutable. Is there an appetite for making this a mutable type? It would allow for convenient in place statements like 

`Vec2.heading = pi / 2` 
`Vec2.heading += angle`

which I think is an elegant way of working with them and a similar thing could be achieved with the magnitude property. Currently to add friction, say, is 

`velocity_vector = velocity_vector.scale(friction_strength)`

which could be replaced with an __imul__ 

`vel *= friction_strength`

...after sorting multiplying between vectors and scalars

Reply:
There's been a lot of profiling to make these types fast. All types extend `tuple` for that reason. We only have a few milliseconds per frame, so this is important 😃